src/common/types.ts

```diff
@@ -0,0 +1,16 @@
+import { BaseItem, CleanupItem } from './interfaces';
+
+type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
+  {
+    [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>;
+  }[Keys];
+
+export type AsyncFunc<T = unknown> = () => Promise<T>;
+
+export type ItemId = string | symbol;
+
+export type RegisterOptions = Partial<CleanupItem> & Pick<CleanupItem, 'func'>;
+
+export type RemoveItem = RequireAtLeastOne<BaseItem>;
+
+export type FinishStatus = 'success' | 'timedout' | 'preFailed' | 'postFailed';
```

Comment:
I think we should rename `timedout` to `timeout`, a timeless status name

Reply:
done.